packages/assets-controllers/src/TokensController.ts

```diff
@@ -700,49 +704,84 @@ export class TokensController extends BaseController<
    * Parameters will be validated according to the asset type being watched.
    *
    * @param options - The method options.
-   * @param options.asset - The asset to be watched. For now only ERC20 tokens are accepted.
    * @param options.type - The asset type.
+   * @param options.address - The contract address of the asset.
+   * @param options.image - An optional URL to the asset's icon.
    * @param options.interactingAddress - The address of the account that is requesting to watch the asset.
    * @param options.networkClientId - Network Client ID.
-   * @returns Object containing a Promise resolving to the suggestedAsset address if accepted.
+   * @returns A promise that resolves if the asset was watched successfully, and rejects otherwise.
    */
   async watchAsset({
-    asset,
     type,
+    address,
+    image,
     interactingAddress,
     networkClientId,
   }: {
-    asset: Token;
     type: string;
+    address: string;
+    image: string;
```

Comment:
My thinking here: Want a clean interface for what dapps/callers provide, with a distintion from the data we supply.  Don't want to pass a `Token` because it contains fields users shouldn't be able to set.  Don't want a bug where we forget to clear those fields.